ND4J Initialization fails with Access Control Exception for "getenv.*"

#### Issue Description

While initializing ND4J, I see runtime exception stating access was denied to getenv.*

By adding Runtime permission to getenv.*, it gives access to ALL environment variables which is not very secure.

Expected Behavior: Exceptions can be thrown asking for access permissions to access specific environment variables instead of all env variables.

Exception:
Caused by: java.lang.RuntimeException: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "getenv.*")
	at org.nd4j.linalg.factory.Nd4j.initWithBackend(Nd4j.java:6212)
	at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:6087)
	at org.nd4j.linalg.factory.Nd4j.<clinit>(Nd4j.java:201)
	... 43 more
Caused by: java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "getenv.*")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:884)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at java.lang.System.getenv(System.java:944)
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.<init>(NativeOpExecutioner.java:73)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.nd4j.linalg.factory.Nd4j.initWithBackend(Nd4j.java:6185)
	... 45 more

#### Version Information
0.9.1

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 0.9.1
* platform information (OS, etc): macOS Sierra 10.12.6, Eclipse IDE, JDK 1.8.0_121
* CUDA version, if used - I am not using CUDA
* NVIDIA driver version, if in use - N/A

--- a/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/jita/conf/Configuration.java
+++ b/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/jita/conf/Configuration.java
@@ -209,96 +209,96 @@ public class Configuration implements Serializable {
 
 
     private void parseEnvironmentVariables() {
-        Map<String, String> env = System.getenv();
 
-        if (env.containsKey(MAX_BLOCK_SIZE)) {
+        // Do not call System.getenv(): Accessing all variables requires higher security privileges
+        if (System.getenv(MAX_BLOCK_SIZE) != null) {
             try {
-                int var = Integer.parseInt(env.get(MAX_BLOCK_SIZE));
+                int var = Integer.parseInt(System.getenv(MAX_BLOCK_SIZE));
                 setMaximumBlockSize(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", MAX_BLOCK_SIZE, env.get(MAX_BLOCK_SIZE));
+                log.error("Can't parse {}: [{}]", MAX_BLOCK_SIZE, System.getenv(MAX_BLOCK_SIZE));
             }
         }
 
-        if (env.containsKey(MIN_BLOCK_SIZE)) {
+        if (System.getenv(MIN_BLOCK_SIZE) != null) {
             try {
-                int var = Integer.parseInt(env.get(MIN_BLOCK_SIZE));
+                int var = Integer.parseInt(System.getenv(MIN_BLOCK_SIZE));
                 setMinimumBlockSize(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", MIN_BLOCK_SIZE, env.get(MIN_BLOCK_SIZE));
+                log.error("Can't parse {}: [{}]", MIN_BLOCK_SIZE, System.getenv(MIN_BLOCK_SIZE));
             }
         }
 
-        if (env.containsKey(MAX_GRID_SIZE)) {
+        if (System.getenv(MAX_GRID_SIZE) != null) {
             try {
-                int var = Integer.parseInt(env.get(MAX_GRID_SIZE));
+                int var = Integer.parseInt(System.getenv(MAX_GRID_SIZE));
                 setMaximumGridSize(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", MAX_GRID_SIZE, env.get(MAX_GRID_SIZE));
+                log.error("Can't parse {}: [{}]", MAX_GRID_SIZE, System.getenv(MAX_GRID_SIZE));
             }
         }
 
-        if (env.containsKey(DEBUG_ENABLED)) {
+        if (System.getenv(DEBUG_ENABLED) != null) {
             try {
-                boolean var = Boolean.parseBoolean(env.get(DEBUG_ENABLED));
+                boolean var = Boolean.parseBoolean(System.getenv(DEBUG_ENABLED));
                 enableDebug(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", DEBUG_ENABLED, env.get(DEBUG_ENABLED));
+                log.error("Can't parse {}: [{}]", DEBUG_ENABLED, System.getenv(DEBUG_ENABLED));
             }
         }
 
-        if (env.containsKey(FORCE_SINGLE_GPU)) {
+        if (System.getenv(FORCE_SINGLE_GPU) != null) {
             try {
-                boolean var = Boolean.parseBoolean(env.get(FORCE_SINGLE_GPU));
+                boolean var = Boolean.parseBoolean(System.getenv(FORCE_SINGLE_GPU));
                 allowMultiGPU(!var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", FORCE_SINGLE_GPU, env.get(FORCE_SINGLE_GPU));
+                log.error("Can't parse {}: [{}]", FORCE_SINGLE_GPU, System.getenv(FORCE_SINGLE_GPU));
             }
         }
 
-        if (env.containsKey(VERBOSE)) {
+        if (System.getenv(VERBOSE) != null) {
             try {
-                boolean var = Boolean.parseBoolean(env.get(VERBOSE));
+                boolean var = Boolean.parseBoolean(System.getenv(VERBOSE));
                 setVerbose(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", VERBOSE, env.get(VERBOSE));
+                log.error("Can't parse {}: [{}]", VERBOSE, System.getenv(VERBOSE));
             }
         }
 
-        if (env.containsKey(USE_PREALLOCATION)) {
+        if (System.getenv(USE_PREALLOCATION) != null) {
             try {
-                boolean var = Boolean.parseBoolean(env.get(USE_PREALLOCATION));
+                boolean var = Boolean.parseBoolean(System.getenv(USE_PREALLOCATION));
                 allowPreallocation(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", USE_PREALLOCATION, env.get(USE_PREALLOCATION));
+                log.error("Can't parse {}: [{}]", USE_PREALLOCATION, System.getenv(USE_PREALLOCATION));
             }
         }
 
-        if (env.containsKey(MAX_DEVICE_CACHE)) {
+        if (System.getenv(MAX_DEVICE_CACHE) != null) {
             try {
-                long var = Long.parseLong(env.get(MAX_DEVICE_CACHE));
+                long var = Long.parseLong(System.getenv(MAX_DEVICE_CACHE));
                 setMaximumDeviceCache(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", MAX_DEVICE_CACHE, env.get(MAX_DEVICE_CACHE));
+                log.error("Can't parse {}: [{}]", MAX_DEVICE_CACHE, System.getenv(MAX_DEVICE_CACHE));
             }
         }
 
 
-        if (env.containsKey(MAX_HOST_CACHE)) {
+        if (System.getenv(MAX_HOST_CACHE) != null) {
             try {
-                long var = Long.parseLong(env.get(MAX_HOST_CACHE));
+                long var = Long.parseLong(System.getenv(MAX_HOST_CACHE));
                 setMaximumHostCache(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", MAX_HOST_CACHE, env.get(MAX_HOST_CACHE));
+                log.error("Can't parse {}: [{}]", MAX_HOST_CACHE, System.getenv(MAX_HOST_CACHE));
             }
         }
 
-        if (env.containsKey(MAX_DEVICE_ALLOCATION)) {
+        if (System.getenv(MAX_DEVICE_ALLOCATION) != null) {
             try {
-                long var = Long.parseLong(env.get(MAX_DEVICE_ALLOCATION));
+                long var = Long.parseLong(System.getenv(MAX_DEVICE_ALLOCATION));
                 setMaximumSingleDeviceAllocation(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", MAX_DEVICE_ALLOCATION, env.get(MAX_DEVICE_ALLOCATION));
+                log.error("Can't parse {}: [{}]", MAX_DEVICE_ALLOCATION, System.getenv(MAX_DEVICE_ALLOCATION));
             }
         }
 

--- a/nd4j-backends/nd4j-backend-impls/nd4j-native/src/main/java/org/nd4j/linalg/cpu/nativecpu/ops/NativeOpExecutioner.java
+++ b/nd4j-backends/nd4j-backend-impls/nd4j-native/src/main/java/org/nd4j/linalg/cpu/nativecpu/ops/NativeOpExecutioner.java
@@ -74,23 +74,22 @@ public class NativeOpExecutioner extends DefaultOpExecutioner {
     public NativeOpExecutioner() {
         tadManager.init(loop, constantHandler);
 
-        Map<String, String> env = System.getenv();
-
-        if (env.containsKey(DEBUG_ENABLED)) {
+        // Do not call System.getenv(): Accessing all variables requires higher security privileges
+        if (System.getenv(DEBUG_ENABLED) != null) {
             try {
-                boolean var = Boolean.parseBoolean(env.get(DEBUG_ENABLED));
+                boolean var = Boolean.parseBoolean(System.getenv(DEBUG_ENABLED));
                 loop.enableDebugMode(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", DEBUG_ENABLED, env.get(DEBUG_ENABLED));
+                log.error("Can't parse {}: [{}]", DEBUG_ENABLED, System.getenv(DEBUG_ENABLED));
             }
         }
 
-        if (env.containsKey(VERBOSE)) {
+        if (System.getenv(VERBOSE) != null) {
             try {
-                boolean var = Boolean.parseBoolean(env.get(VERBOSE));
+                boolean var = Boolean.parseBoolean(System.getenv(VERBOSE));
                 loop.enableVerboseMode(var);
             } catch (Exception e) {
-                log.error("Can't parse {}: [{}]", VERBOSE, env.get(VERBOSE));
+                log.error("Can't parse {}: [{}]", VERBOSE, System.getenv(VERBOSE));
             }
         }
     }

--- a/nd4j-context/src/main/java/org/nd4j/linalg/factory/Nd4jBackend.java
+++ b/nd4j-context/src/main/java/org/nd4j/linalg/factory/Nd4jBackend.java
@@ -250,7 +250,8 @@ public abstract class Nd4jBackend {
         String[] jarUris;
         if (System.getProperties().containsKey(DYNAMIC_LOAD_CLASSPATH_PROPERTY) && !triedDynamicLoad) {
             jarUris = System.getProperties().getProperty(DYNAMIC_LOAD_CLASSPATH_PROPERTY).split(";");
-        } else if (System.getenv().containsKey(DYNAMIC_LOAD_CLASSPATH) && !triedDynamicLoad) {
+        // Do not call System.getenv(): Accessing all variables requires higher security privileges
+        } else if (System.getenv(DYNAMIC_LOAD_CLASSPATH) != null && !triedDynamicLoad) {
             jarUris = System.getenv(DYNAMIC_LOAD_CLASSPATH).split(";");
         }
 

